### PR TITLE
Java client issues batch 2

### PR DESCRIPTION
**Pull request by l-trotta**
Fixes provided in this PR: 

- [744](https://github.com/elastic/elasticsearch-java/issues/744), [376](https://github.com/elastic/elasticsearch-java/issues/376): [ExecuteEnrichPolicyResponse](https://github.com/elastic/elasticsearch-specification/blob/main/specification/enrich/execute_policy/ExecuteEnrichPolicyResponse.ts) should have `status` optional, otherwise when called with `waitForCompletion=false` the parsing will fail
- [317](https://github.com/elastic/elasticsearch-java/issues/317): not really the issue described (that one was already fixed), but [`transient_metadata`](https://github.com/elastic/elasticsearch-specification/blob/main/specification/security/get_role/types.ts#L34) should be a map, not a single field class. Starting a basic elasticsearch instance with `xpack.security.enabled` and a password and trying to call `client.security().getRole()` will otherwise fail. 
- [379](https://github.com/elastic/elasticsearch-java/issues/379): `*_size_bytes` fields should be `long`, not `integer`
- [394](https://github.com/elastic/elasticsearch-java/issues/394): [GeoPoint](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_types/mapping/geo.ts#L23) is missing [fields](https://www.elastic.co/guide/en/elasticsearch/reference/current/geo-point.html#geo-point-params)
- [403](https://github.com/elastic/elasticsearch-java/issues/403): `source` in [InlineGet](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_types/common.ts#L321) should be [optional](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/get/GetResult.java#L79)
- [409](https://github.com/elastic/elasticsearch-java/issues/409): `stored_field` in [InnerHits](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_global/search/_types/hits.ts#L106) should be [`stored_fields` ](https://github.com/elastic/elasticsearch/blob/main/server/src/main/java/org/elasticsearch/index/query/InnerHitBuilder.java#L143)
- [450](https://github.com/elastic/elasticsearch-java/issues/450): [RankFeaturesProperty](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_types/mapping/core.ts#L189) should have `positive_score_impact` like [RankFeatureProperty does](https://www.elastic.co/guide/en/elasticsearch/reference/current/rank-features.html)
- [754](https://github.com/elastic/elasticsearch-java/issues/754): `dims` should be [optional](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html#dense-vector-params) for [dense vectors](https://github.com/elastic/elasticsearch-specification/blob/main/specification/_types/mapping/complex.ts#L51)

### CODE REVIEW

**Review by Anaethelion — CHANGES_REQUESTED** [review]
Unsure about `GeoPointProperty`, it seems to fit the server description, maybe these should be a level higher.

File: specification/_types/mapping/core.ts [hunk 1]
```diff
@@ -187,6 +187,7 @@ export class RankFeatureProperty extends PropertyBase {
 }
 
 export class RankFeaturesProperty extends PropertyBase {
+  positive_score_impact?: boolean
```
[the remark this answers is not in the document]
Comment (swallez) [reply specification/_types/mapping/core.ts:190]: I'll keep it optional. It's [represented in the code](https://github.com/elastic/elasticsearch/blob/d6f91d9b21eed2b7a0df412e82a232f96efff70a/modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/RankFeaturesFieldMapper.java#L47) as a `Parameters` object that will return its default value if not set.

File: specification/security/get_role/types.ts [hunk 2]
```diff
@@ -21,17 +21,17 @@ import {
...
 import { RoleTemplate } from '@security/_types/RoleTemplate'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
 
 export class Role {
   cluster: string[]
   indices: IndicesPrivileges[]
   metadata: Metadata
   run_as: string[]
-  transient_metadata: TransientMetadataConfig
+  transient_metadata?: Dictionary<string, UserDefinedValue>
```
[the remark this answers is not in the document]
Comment (swallez) [reply specification/security/get_role/types.ts:34]: Yes. Also remove `TransientMetadataConfig` that is now useless.

[the remark this answers is not in the document]
Comment (Anaethelion) [reply specification/security/get_role/types.ts:34]: Agreed.

**Review by Anaethelion — APPROVED** [review]

**Pull request merged**